meeting on 2023-02-17 (#50)
ElektraInitiative/PermaplanT: doc/meetings/2023_02_17.md

# Meeting 2023-02-17

https://github.com/ElektraInitiative/PermaplanT/issues/37#issuecomment-1433926689

## Questions from Lukas and Markus:

1. Should we pin the rust version in any way, or is it ok to always use the latest version?

>     - we use latest

Reviewer: ```suggestion
    - we use Rust version 1.67.1 (2023-02-09)
```

please also update doc/decisions/backend_programming_language.md and link there.

>     - we use Rust version 1.67.1 (2023-02-09) [(link to the decision)](https://github.com/ElektraInitiative/PermaplanT/blob/meeting-17.02.2023/doc/decisions/backend_programming_language.md)

Reviewer: I would simply use internal links, they should at least work within GitHub:

```suggestion
    - we use Rust version 1.67.1 (2023-02-09), [see decision](/doc/decisions/backend_programming_language.md)
```

2. How should we handle the situation during build? (no Postgres, etc.)
 
3. How do we inject the correct `BASE_URL` for the backend into the frontend?

>     - .env file

Reviewer: please link to issue #49

## Meeting notes:

Updates from Markus:

> - there should be easy ways to indicate areas of interest which do not fit any layer

Reviewer: ```suggestion
- there should be easy ways to indicate areas of interest, e.g. by crosshatching areas which do not have a dedicated symbol in the landscape layer
```

- gardens of various sizes for beta testing

> - Yvonne started drawing for permaplant

Reviewer: ```suggestion
- Yvonne started drawing for PermaplanT
```

> - Next friday set milestones with final dates

Reviewer: ```suggestion
- Next Friday set milestones with final dates
```

Discussions:

- Deployment in react https://create-react-app.dev/docs/deployment/
- Both deployment methods are equal from our point of view. /api proxy vs static port

> - treffle.io is a big database but not much information

Reviewer: ```suggestion
- treffle.io is a big database but is lacking vital information for common gardening plants
```

> - Diesel needs a database even on the compile time

Reviewer: ```suggestion
- Diesel needs a database even on compile time
```

> - 3 independent PSQL containers for PR, main and releases

Reviewer: ```suggestion
- 3 independent containers for PRs, master and releases
```

Author: I've taken the suggestions over, however, the word "containers" is ambiguous in my opinion e.g. docker is the first thing that comes to my mind. I suggest we rephrase or specify it in the future to avoid misunderstandings among the new members of the project

Reviewer: Good idea! Specifically, we use the lightweight containers from Proxmox VE, which use [Linux Containers (LXC)](https://linuxcontainers.org/lxc/introduction/) see also https://pve.proxmox.com/wiki/Linux_Container

> - 3 independent containers for PRs, master and releases

Reviewer: ```suggestion
- 3 independent LXC containers for PRs, master and releases [see issue](https://github.com/ElektraInitiative/PermaplanT/issues/53)
```
